Nodejs/Product/Nodejs/Project/NodejsProjectNode.cs

```diff
@@ -202,7 +201,11 @@ public NodejsProjectNode(NodejsProjectPackage package)
 
         public bool IsTypeScriptProject {
             get {
-                return string.Equals(GetProjectProperty(NodejsConstants.EnableTypeScript), "true", StringComparison.OrdinalIgnoreCase);
+                var task = ProjectMgr.Site.GetUIThread().InvokeAsync(() => {
+                    return string.Equals(GetProjectProperty(NodejsConstants.EnableTypeScript), "true", StringComparison.OrdinalIgnoreCase);
+                });
+                task.Wait();
+                return task.Result;
```

Comment:
Also, an we check on the other callers here? Are they on the UI thread? If so, we should also assert that we're on the UI thread here so we don't run into issues like this in the future.